Code/GraphMol/ChemTransforms/ChemTransforms.cpp

```diff
@@ -51,6 +51,18 @@ void updateSubMolConfs(const ROMol &mol, RWMol &res,
     res.addConformer(newConf, false);
   }
 }
+
+struct SideChainMapping {
+  int molIndex;
+  int coreIndex = -1;
+  bool useMatch;
+
+  SideChainMapping(int molIndex)
+      : molIndex(molIndex), coreIndex(-1), useMatch(false) {}
+  SideChainMapping(int molIndex, int coreIndex, bool useMatch)
+      : molIndex(molIndex), coreIndex(coreIndex), useMatch(useMatch) {}
+};
+
 }  // namespace
 
 ROMol *deleteSubstructs(const ROMol &mol, const ROMol &query, bool onlyFrags,
```

Comment:
There need to be tests for this new behavior I think.

Reply:
The only time the new behavior is activated is when RGD has two attachment points mapped to a single atom.  That is tested in testSingleAtomBridge in testRGroupDecomp.cpp.  I think that testRGroupDecomp.cpp is the right place to test the new behavior.  Admittedly, I only have a single test, though it does check 5 decompositions.

My bigger concern is that somehow I have broken the existing behavior of replaceCore- hopefully the existing tests protect against this.

Reply:
> My bigger concern is that somehow I have broken the existing behavior of replaceCore- hopefully the existing tests protect against this.

I think it's fine to rely on this.

Reply:
But I agree with @bp-kelley that there should be a test in ChemTransforms for the new behavior.
It's good to also have the tests in the RGD code that demonstrate that your changes work there, but having more of a unit-test-like test here would also be good.

Reply:
I'll put together a ChemTransforms test